### PR TITLE
[5.0] URLSessionTask: implement InputStream.

### PR DESCRIPTION
**Pull request by albertaleksieiev**
>Allows using InputStream in _URLRequest_ as `httpBodyStream`. Also, implement `seekInputStream` in _NativeProtocol_.

This is a cherry-pick of the #1629

### CODE REVIEW

**Comment by albertaleksieiev** [comment]
@ianpartridge can you please trigger a CI, thanks!

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by millenomi** [comment]
@ianpartridge good to merge?

File: Foundation/Stream.swift [hunk 1]
```diff
@@ -114,7 +114,10 @@ open class Stream: NSObject {
 // InputStream is an abstract class representing the base functionality of a read stream.
 // Subclassers are required to implement these methods.
 open class InputStream: Stream {
-
+    enum _Error: Error {
```
Comment (millenomi) [inline Foundation/Stream.swift]: We don't prefix internal types, typically. `enum Error: Swift.Error`?

File: TestFoundation/TestStream.swift [hunk 2]
```diff
@@ -7,6 +7,30 @@
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 
+#if (os(Linux) || os(Android))
```
Comment (millenomi) [inline TestFoundation/TestStream.swift]: Please guard this block and tests that use internal members with `#if NS_FOUNDATION_ALLOWS_TESTABLE_IMPORT`. It's not always available.

**Comment by millenomi** [comment]
I have to request a change in tests, but LGTM otherwise if @ianpartridge is okay with the implementation.

**Comment by albertaleksieiev** [comment]
Thanks for your review @millenomi, I'll fix it ASAP 👍

**Comment by spevans** [comment]
@swift-ci test

**Comment by spevans** [comment]
@swift-ci test

**Comment by albertaleksieiev** [comment]
@millenomi please check my latest changes, thanks!

**Comment by spevans** [comment]
@millenomi is this ok to merge now?

**Comment by albertaleksieiev** [comment]
@millenomi is this good now after my changes?

**Comment by millenomi** [comment]
cc @ianpartridge I'm going to take this barring significant issues; let me know if we need a revert.

@albertaleksieiev do you have a PR for master?

**Pull request merged**